src/frontend/src/lib/services/auth-broadcast.services.ts

```diff
@@ -0,0 +1,60 @@
+import { OISY_URL } from '$lib/constants/oisy.constants';
+import { authStore } from '$lib/stores/auth.store';
+
+export const AUTH_BROADCAST_CHANNEL = 'oisy_auth_channel';
+
+export const AUTH_BROADCAST_MESSAGE_LOGIN_SUCCESS = 'authClientLoginSuccess';
+
+export interface AuthBroadcastChannelInterface {
+	init: () => void;
+	close: () => void;
+	postLoginSuccess: () => void;
+}
+
+// If the user has more than one tab open in the same browser,
+// there could be a mismatch of the cached delegation chain vs the identity key of the `authClient` object.
+// This causes the `authClient` to be unable to correctly sign calls, raising Trust Errors.
+// To mitigate this, we use a `BroadcastChannel` to notify other tabs when a login has occurred, so that they can sync their `authClient` object.
+class AuthBroadcastChannel implements AuthBroadcastChannelInterface {
+	private bc: BroadcastChannel;
+
+	constructor() {
+		this.bc = new BroadcastChannel(AUTH_BROADCAST_CHANNEL);
+	}
+
+	private onLoginSuccess = (handler: () => Promise<void>) => {
+		this.bc.onmessage = async (event) => {
+			if (event.origin === OISY_URL && event.data === AUTH_BROADCAST_MESSAGE_LOGIN_SUCCESS) {
```

Comment:
IMO asserting the `origin` is the most important feature. Do we have a test for it?

Reply:
Great